test/k23si/QueryTest.cpp

```diff
@@ -248,10 +273,18 @@ seastar::future<> runScenario01() {
     });
 }
 
-// Query conflict cases
+// Query projection cases
 seastar::future<> runScenario02() {
     K2INFO("runScenario02");
 
+    K2INFO("Projection reads for giving fields");
+    return doQuery("a", "c", -1, false, 2, 1, k2::dto::K23SIStatus::OK, {"partition"});
```

Comment:
Like I mentioned above, let's check that the data is correct. Also, can you add a test that tries to project a field that is not part of the schema?